scripts/bundle-cjs.mjs

```diff
@@ -0,0 +1,71 @@
+#!/usr/bin/env node
+
+import { resolve } from 'node:path';
+import * as esbuild from 'esbuild';
+import packageJson from '../package.json' with {'type': 'json'};
+
+// import.meta.dirname is not available before Node 20
+const __dirname = import.meta.dirname;
+
+const mainEntrypoint = resolve(__dirname, '../src/index.ts');
+const outDir = resolve(__dirname, '../dist');
+
+// Those dependencies are distributed as ES modules, and cannot be externalized
+// in our CJS bundle. We convert them to CJS and bundle them instead.
+const bundledDeps = [
+  'axios',
+  'form-data',
+  'isomorphic-ws',
+  'base64-js',
+  // 'jsonwebtoken', 'crypto', 'https'
+];
+
+const deps = Object.keys({
+  ...packageJson.dependencies,
+  ...packageJson.peerDependencies,
+});
+const external = deps.filter((dep) => !bundledDeps.includes(dep));
+
+/** @type esbuild.BuildOptions */
+const cjsBundleConfig = {
+  entryPoints: [mainEntrypoint],
+  bundle: true,
+  format: 'cjs',
+  target: 'es2015',
+  external,
+  outdir: outDir,
+  outExtension: { '.js': '.cjs' },
+  sourcemap: 'linked',
+};
+
+const browserPlugin = {
+  name: 'ignoreBrowser',
+  setup: (build) => {
+    const options = build.initialOptions;
+
+    if (options.platform !== 'browser') return;
+
+    build.onResolve({ filter: /(node:.+)|(jsonwebtoken)/ }, async ({ path }) => ({ path, external: true }));
+  },
+};
```

Comment:
I'm not sure if this is the right approach, previous builds seemed to keep the imports (of `crypto`, `https` and `jsonwebtoken`) even though they shouldn't be there in my opinion.

Reply:
Nevermind, [this](https://github.com/GetStream/stream-chat-js/pull/1468/files#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519R22-R26) seems to work just as well. :)